### ISSUE TITLE
Update va-combo-box focus

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/vets-design-system-documentation#5705]
## Description
Based on this issue [Staging Review finding: Using a screen reader, unable to select combobox options](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5127), the va-combo-box component needs to be updated to allow for focus to move to the results. 

## Details
Suggestions from Jason Day from Governance team. 

> Some options:
> 
> put the results list in the DOM before the close "x", so that swiping would move from the input to the results list container
> Or, when the user activates the caret, put focus onto the search results container (currently focus resets to the input, and when swiping, focus moves to the "x", closing the results

## Tasks
- [ ] Review the staging review issue with details of their findings
- [ ] Update combo box component to address the issue
- [ ] Get DS folks to review (accessibility, engineering and design if needed)
- [ ] Publish update 

## Acceptance Criteria
- [ ] Combo box component updated to address this SR finding
